enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,291 @@
+
+
+### API Extensions
+
+The OpenShift Builds will intoroduce a new configuration CRD, `OpenShiftBuild`. It will contain 
+`spec` fields that configure the Shared Resource CSI driver in addition to Shipwright-related
+components. The operator will also deploy (directly or indirectly) the following:
+
+- Custom resources for the Shipwright Operator
+- Custom resources and webhooks for Shipwright Build
+- Custom resources and webhooks for the Shared Resource CSI Driver
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+As noted above, the operator will not deploy the Shared Resource CSI Driver if the cluster has
+the respective OCP feature gate enabled. This generally applies to tech preview clusters or
+clusters with custom fature gate tunings.
+
+With the start of OCP 4.16, the Shared Resource CSI driver operator will be removed from the
+[Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator). The feature
+will likewise be removed from OpenShift's feature sets - any code that needs the most up to date
+set of feature sets will need to be updated (ex - openshift/library-go).
+The Shared Resource CSI driver itself will need to be removed from the OpenShift payload. Its
+components will need to be re-productized as an OLM operand.
+
+Notably, the design of this operator does **_not_** intend to use OLM API dependency resolution
+to automate the deployment and management of OpenShift Pipelines. Cluster admimins will need to
+separately (and manually) install OpenShift Pipelines.
+
+
+### Risks and Mitigations
+
+#### Duplicate CSI Drivers
+
+On a tech preview cluster, the Same CSI driver could be deployed twice on tech preview clusters. The operator will check if the `CSIDriverSharedResource` feature is enabled before
+deploying the Shared Resource CSI driver itself.
+
+
+#### Unwanted components/CRDs
+
+Admins or users may want to use the Shared Resource CSI Driver without Shipwright or Tekton - for
+example, developers or teams that heavily rely on the `BuildConfig` APIs today. To address this
+concern, the OpenShift Builds operator will not rely on OLM API resolution to automate the
+installation and deployment of OpenShift Pipelines. The operator will instead report an appropriate
+status message if `spec.shipwright.build.enabled` is set to `true` and it detects that OpenShift
+Pipelines has not been installed.
+
+
+### Drawbacks
+
+- Shared resources no longer a part of OpenShift.
+- Customers who need entitlements in builds will need to install a new operator, or resort to
+  current cumbersome work-arounds.
+- Potential impact on Red Hat components that need shared resources and/or RHEL entitlements.
+- Manual installation of OpenShift Pipelines is required to deploy Shipwright Builds. Shipwright's
+  upstream operator currently installs Tekton via [OLM API dependency resolution](https://olm.operatorframework.io/docs/concepts/olm-architecture/dependency-resolution/).
```

Comment:
to be honest i think avoiding the specifying of olm dependencies is a safer choice right now anyway.  That may(hopefully will) change with olm v1, but olm v0 dependencies have significant challenges.